avro.go

```diff
@@ -0,0 +1,102 @@
+package difftable
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/linkedin/goavro"
+)
+
+type avroRow struct {
+	record map[string]interface{}
+}
+
+func (r *avroRow) Bytes(col string) []byte {
+	val := r.record[col]
+	return []byte(fmt.Sprint(val))
+}
+
+func (r *avroRow) Value(col string) interface{} {
+	return r.record[col]
+}
+
+type avroTable struct {
+	rdr    *goavro.OCFReader
+	key    []string
+	cols   map[string]string
+	record map[string]interface{}
+}
+
+func (a *avroTable) Key() []string {
+	return a.key
+}
+
+func (a *avroTable) Cols() map[string]string {
+	return a.cols
+}
+
+func (a *avroTable) Row() Row {
+	return &avroRow{
+		record: a.record,
+	}
+}
+
+func (a *avroTable) Next() (bool, error) {
+	if !a.rdr.Scan() {
+		return false, a.rdr.Err()
+	}
+
+	datum, err := a.rdr.Read()
+	if err != nil {
+		return false, err
+	}
+
+	a.record = datum.(map[string]interface{})
+
+	return true, nil
+}
+
+func AvroTable(rdr *goavro.OCFReader, key []string, renames map[string]string) (Table, error) {
+	for i, k := range key {
+		if n, ok := renames[k]; ok {
+			key[i] = n
+		}
+	}
+
+	var m map[string]interface{}
+	err := json.Unmarshal([]byte(rdr.Codec().Schema()), &m)
+	if err != nil {
+		return nil, err
+	}
+
+	if t, ok := m["type"].(string); !ok || t != "record" {
+		return nil, fmt.Errorf("record schema required, got %s", t)
+	}
+
+	fields, ok := m["fields"].([]interface{})
+	if !ok {
+		return nil, errors.New("could not parse record fields")
+	}
+
+	cols := make(map[string]string)
+	for _, x := range fields {
+		f := x.(map[string]interface{})
```

Comment:
Is there a reason why you're not catching a potential failed type assertion here?

Reply:
I should add a check for consistency, however fields have a known structure per the Avro spec. Since I know the schema is a record (per the above logic), I know that fields are encoded as maps and have specific keys.

Reply:
Right, that makes sense, maybe put what you wrote here in a comment for readers who are not familiar with the avro spec.